indexer/drawing_rule_def.cpp

```diff
@@ -53,4 +53,25 @@ void MakeUnique(KeysT & keys)
   sort(keys.begin(), keys.end(), less_key());
   keys.resize(distance(keys.begin(), unique(keys.begin(), keys.end(), equal_key())));
 }
+
+double CalcAreaBySizeDepth(FeatureType & f)
+{
+  // Calculate depth based on areas' bbox sizes instead of style-set priorities.
+  m2::RectD const r = f.GetLimitRectChecked();
+  // Raw areas' size range is about (1e-11, 3000).
+  double const areaSize = r.SizeX() * r.SizeY();
+  // Use log2() to have more precision distinguishing smaller areas.
+  double const areaSizeCompact = std::log2(areaSize);
```

Comment:
My fix is missing. Happens that areaSize is 0 -> areaSizeCompact is NAN.

Reply:
yeap lost in a rebase
I'll fix it.